pkgs/development/libraries/qt-6/modules/qtshadertools.nix

```diff
@@ -1,8 +1,18 @@
 { qtModule
 , qtbase
+, stdenv
+, lib
+, pkgsBuildBuild
 }:
 
+let
+  isCrossBuild = stdenv.hostPlatform != stdenv.buildPlatform;
+in
 qtModule {
   pname = "qtshadertools";
   propagatedBuildInputs = [ qtbase ];
+  cmakeFlags = [ ] ++ lib.optionals isCrossBuild [
+    "-DQt6ShaderToolsTools_DIR=${pkgsBuildBuild.qt6.qtshadertools}/lib/cmake/Qt6ShaderToolsTools"
```

Comment:
Is the isCrossBuild conditional just to avoid mass rebuild?

Reply:
Also to avoid infinite recursion.